### PR TITLE
CSP-1354 matched paye aorn and email not linked shutter page

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.Provider.PR.Web/Controllers/AddEmployer/SearchByPayeController.cs
Comment (DevenKShah) [inline src/SFA.DAS.Provider.PR.Web/Controllers/AddEmployer/SearchByPayeController.cs:1]: - The outer api response should not be nullable
- Rename `hasExistingAccount` to `accountDoesNotExist`
- Move this above the Single legal entity check

File: src/SFA.DAS.Provider.PR.Web/Models/AddEmployer/PayeAornMatchedEmailNotLinkedViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+﻿namespace SFA.DAS.Provider.PR.Web.Models.AddEmployer;
+
+public record PayeAornMatchedEmailNotLinkedViewModel(string EmployerName, string PayeReference, string Aorn, string Email, string CancelLink);
```
Comment (DevenKShah) [inline src/SFA.DAS.Provider.PR.Web/Models/AddEmployer/PayeAornMatchedEmailNotLinkedViewModel.cs:3]: Convert this into a class

File: src/SFA.DAS.Provider.PR.Web/Controllers/AddEmployer/SearchByPayeController.cs
Comment (MarkFCain) [reply src/SFA.DAS.Provider.PR.Web/Controllers/AddEmployer/SearchByPayeController.cs:1]: ok

File: src/SFA.DAS.Provider.PR.Web/Controllers/AddEmployer/SearchByPayeController.cs [hunk 2]
```diff
@@ -169,6 +179,55 @@ public IActionResult PayeAornShutterPage([FromRoute] int ukprn)
...
+        {
+            sessionModel.PermissionToAddCohorts = SetPermissions.AddRecords.Yes;
+            _sessionService.Set(sessionModel);
+        }
+
+        if (sessionModel.PermissionToRecruit == null)
+        {
+            sessionModel.PermissionToRecruit = SetPermissions.RecruitApprentices.Yes;
+            _sessionService.Set(sessionModel);
+        }
```
Comment (DevenKShah) [inline src/SFA.DAS.Provider.PR.Web/Controllers/AddEmployer/SearchByPayeController.cs:225]: This is already being taken care of on the `AddPermissionsAndEmployer` so it is redundant here. Also there is no need to have this POST action, instead the `Continue` can be directly linked to that page.

**Review by DevenKShah — DISMISSED** [review]

**Pull request merged**